lib/matplotlib/axes/_axes.py

```diff
@@ -4457,13 +4457,6 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
 
         scales = s   # Renamed for readability below.
 
-        # to be API compatible
-        if verts is not None:
```

Comment:
`verts` should be be removed from the method signature. Unfortunately, strictly speaking this would be a breaking change again because you change the position of `edgecolors`. But nevertheless I think that's what we should do - and since `edgecolors` is changing its position anyway, you can move the kwonly marker `*` before it.

The alternative would be to keep `verts` and raise a TypeError here.

Reply:
I added a `_delete_parameter` on `verts`.  Sure, that'll take two more minor versions before removing it from the signature, but heh, there's no urgency either...